libs/ledgerjs/packages/hw-app-eth/tests/ledgerService.unit.test.ts

```diff
@@ -52,6 +52,7 @@ describe("Ledger Service", () => {
         );
 
         expect(resolution).toEqual({
+          domains: [],
```

Comment:
`domains` are not really handled by the `ledgerService` per say, it's just a passthrough to keep the logic of "_everything for clear signing should pass by the `ledgerService`_". 
I'm not a big fan of that, but that's the best solution I got to avoid a very poor DX where people would need to manually add things to the "resolution" after the `resolveTransaction` method was called. 

```typescript 
import Eth, { ledgerService } from "@ledgerhq/hw-app-eth"

const eth = new Eth(transport);
const tx = "e8018504e3b292008252089428ee52a8f3d6e5d15f8b131996950d7f296c7952872bd72a2487400080";
// ↓ this has been provided by the `@ledgerhq/domain-service` resolver at the beginning of the signature flow
const myDomain = { registry: "ens", address: "0x123", domain: "coucou.eth", type: "forward" };
const resolution = await ledgerService.resolveTransaction(tx, null, { erc20: true, nft: true }); 
// ↑ this is supposed to handle the whole clear signing shenanigans
  
// kinda weird to have to do that here
if (youGotADomainToClearSign) {
  resolution.domains = [myDomain];
}

const result = eth.signTransaction("44'/60'/0'/0/0", tx, resolution); 
// Or we would have to add it here as a new parameter in the signTransaction method, 
// which makes it less and less a binding to the nano app + it creates a new signature that 
// might bites us in the ass later if we need yet another parameter
```

VS 

```typescript 
import Eth, { ledgerService } from "@ledgerhq/hw-app-eth"

const eth = new Eth(transport);
const tx = "e8018504e3b292008252089428ee52a8f3d6e5d15f8b131996950d7f296c7952872bd72a2487400080";
// ↓ this has been provided by the `@ledgerhq/domain-service` resolver at the beginning of the signature flow
const myDomain = { registry: "ens", address: "0x123", domain: "coucou.eth", type: "forward" };
const resolution = await ledgerService.resolveTransaction(tx, null, { erc20: true, nft: true, domains: [myDomain] }); 

const result = eth.signTransaction("44'/60'/0'/0/0", tx, resolution); 
```

So to answer the question, no there is no need for new tests there, as `ledgerService.resolveTransaction` is doing nothing with the domains but to keep them as is in the `resolution` variable to make sure `eth.signTransaction` as access to it.

**SIDE NOTE:**
Also why can't `ledgerService.resolveTransaction` get the APDUs itself like with ERC20/NFTs ? 
→ Because it requires a "challenge" provided by the nano app. So the other option would be:

```typescript
import Eth, { ledgerService } from "@ledgerhq/hw-app-eth"

const eth = new Eth(transport);
const tx = "e8018504e3b292008252089428ee52a8f3d6e5d15f8b131996950d7f296c7952872bd72a2487400080";
const myDomain = { registry: "ens", address: "0x123", domain: "coucou.eth", type: "forward" };
const challenge = myDomain ? eth.getChallenge() : undefined; // Requiring the nano to be active already
const resolution = await ledgerService.resolveTransaction(
  tx, 
  null,
  { 
    erc20: true, 
    nft: true, 
    domains: [myDomain], 
    challenge // necessary for sending the domain APDU to the nano app
  }
); 

const result = eth.signTransaction("44'/60'/0'/0/0", tx, resolution); 
```

This means that in our own logic in the `signOperation` Observable, you would already need the nano before the step `device-signature-requested`, which is against what :gre: wants 🤣

Reply:
Neat explanation :ok_hand: thanks man.
Would be very cool in a dedicated doc somewhere... :stuck_out_tongue:

Reply:
Well that 100% the kind of things that would go into the "Explanation" part of a [diataxis](https://diataxis.fr/) documentation for example... If you see what I mean... 😏 :neckbeard:

Reply:
_touché_ :slightly_smiling_face: we're working on the topic with @lvndry , but until then good ol' Confluence is better than nothing.